src/serialization/interval_log/mod.rs

```diff
@@ -683,81 +683,75 @@ fn system_time_as_fp_seconds(time: time::SystemTime) -> f64 {
     }
 }
 
-fn start_time(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], nom::error::ErrorKind)> {
-    map(
-        tuple((
-            tag("#[StartTime: "),
-            fract_sec_duration,
-            char(' '),
-            take_until("\n"),
-            take(1_usize),
-        )),
-        |x| LogEntry::StartTime(x.1),
-    )(input)
+fn start_time(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], ErrorKind)> {
+    let (input, _) = tag("#[StartTime: ")(input)?;
+    let (input, duration) = fract_sec_duration(input)?;
+    let (input, _) = char(' ')(input)?;
+    let (input, _) = take_until("\n")(input)?;
+    let (input, _) = take(1_usize)(input)?;
+    Ok((input, LogEntry::StartTime(duration)))
 }
 
-fn base_time(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], nom::error::ErrorKind)> {
-    map(
-        tuple((
-            tag("#[BaseTime: "),
-            fract_sec_duration,
-            char(' '),
-            take_until("\n"),
-            take(1_usize),
-        )),
-        |x| LogEntry::BaseTime(x.1),
-    )(input)
+fn base_time(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], ErrorKind)> {
+    let (input, _) = tag("#[BaseTime: ")(input)?;
+    let (input, duration) = fract_sec_duration(input)?;
+    let (input, _) = char(' ')(input)?;
+    let (input, _) = take_until("\n")(input)?;
+    let (input, _) = take(1_usize)(input)?;
+    Ok((input, LogEntry::BaseTime(duration)))
 }
 
-fn interval_hist(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], nom::error::ErrorKind)> {
-    map(
-        tuple((
-            opt(map(
-                map_res(
-                    map(tuple((tag("Tag="), take_until(","), take(1_usize))), |p| {
-                        p.1
-                    }),
-                    str::from_utf8,
-                ),
-                |s| Tag(s),
-            )),
-            fract_sec_duration,
-            char(','),
-            fract_sec_duration,
-            char(','),
-            double,
-            char(','),
-            map_res(take_until("\n"), str::from_utf8),
-            take(1_usize),
-        )),
-        |(tag, start_timestamp, _, duration, _, max, _, encoded_histogram, _)| {
-            LogEntry::Interval(IntervalLogHistogram {
-                tag,
-                start_timestamp,
-                duration,
-                max,
-                encoded_histogram,
-            })
-        },
-    )(input)
+fn interval_hist(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], ErrorKind)> {
+    let (input, tag) = opt(map(
+        map_res(
```

Comment:
I'd probably personally write a helper function for the inner parser here, but :man_shrugging: